js/viewer/annotation/base.js

```diff
@@ -6,16 +6,25 @@ function Base(attrs){
     this.isDrawing = false;
     this.constants = attrs.parent.parent.parent.all_config.constants;
     this.urlParams = attrs.parent.parent.parent.parameters;
+    this.annotationUtils = new AnnotationUtils();
+
+    // this is used to make sure that any property that is added to the SVG for display purpose, is not added while saving
+    this.addedProperties = {
+        "vector-effect": true
+    };
     this._attrs = {
```

Comment:
Why did you remove the parts that we were getting the values from `attrs`? Based on what I'm seeing in `attribute-group.js` we're sending `graph-id`. Do you even know why we have `graph-id` as an attribute?

Reply:
I removed that because, but initialization those value were not passed, and were set later during the `set` functions. What I did do is initialize them with default values.

I dont when why we are sending the `graph-id` or why we use it in attributes

Reply:
You should figure it out. With what you did here, you're not using the `graph-id` value that is passed from the `attribute-group.js` (`addAnnotation` function).

If `graph-id` is not needed, you should modify `Path`, `Polyline`, `Polygon`, `Rect`, `Circle`, and `Base` constructor functions to only accept a `parent` (instead of `attrs`).

Reply:
I have added a TODO in the code for the above requirement.